apps/explorer/src/ui/ObjectDetails.tsx

```diff
@@ -0,0 +1,144 @@
+    variants: {
+        size: {
+            small: 'gap-1.25',
+            large: 'gap-2.5',
+        },
+    },
+});
+
+export interface ObjectDetailsProps {
+    id?: string;
+    image: string;
+    name?: string;
+    type: string;
+    nsfw?: boolean;
+    variant: 'small' | 'large';
+}
+
+export function ObjectDetails({
+    id,
+    image,
+    name,
+    type,
+    nsfw = false,
+    variant = 'small',
+}: ObjectDetailsProps) {
+    const [open, setOpen] = useState(false);
+    const close = () => setOpen(false);
+    const openPreview = () => setOpen(true);
+    return (
+        <div className="flex items-center gap-3.75">
+            <LightBox open={open} onClose={close}>
+                <div className="flex flex-col gap-5">
+                    <Image alt={name} src={image} className="rounded-none" />
+                    <Heading variant="heading2/semibold" color="sui-light">
+                        {name}
+                    </Heading>
+                    <Text color="gray-60" variant="body/medium">
+                        {type}
+                    </Text>
+                </div>
+            </LightBox>
+            <div className="relative">
+                <Image
+                    onClick={openPreview}
+                    alt={name}
+                    src={image}
+                    blur={nsfw}
+                    className={imageStyles({ size: variant })}
+                />
+            </div>
+            <div className={textStyles({ size: variant })}>
+                {variant === 'large' ? (
+                    <Heading color="gray-90" variant="heading4/semibold">
+                        {name}
+                    </Heading>
+                ) : (
+                    <Text variant="bodySmall/medium" color="gray-90">
+                        {name}
+                    </Text>
+                )}
+                {id && <ObjectLink objectId={id} />}
+                <Text variant="bodySmall/medium" color="steel-dark">
+                    {type}
+                </Text>
+                {variant === 'large' ? (
+                    <div
+                        onClick={openPreview}
+                        className="mt-2.5 flex cursor-pointer items-center gap-1 text-steel-dark"
+                    >
+                        <Text variant="caption/semibold">Preview</Text>
+                        <ArrowUpRight16 className="inline-block" />
+                    </div>
+                ) : null}
+            </div>
+        </div>
+    );
+}
+
+function FallbackImage({ className }: { className: string }) {
+    return (
+        <div
+            className={clsx(
+                className,
+                'flex items-center justify-center rounded-md bg-gray-45 text-gray-65'
+            )}
+        >
+            <NftTypeImage24 />
+        </div>
+    );
+}
+
+function Image({ className, alt, src, blur = false, onClick }: ImageProps) {
```

Comment:
If we're introducing a base image component can we make this extracted into it's own UI component? We could also improve the loading state and error behavior around this, e.g.: https://github.com/chakra-ui/chakra-ui/blob/main/packages/components/image/src/use-image.ts